### DISCUSSION TITLE
Request: Server option to ignore events from selected device

### DISCUSSION THREAD

**Discussion by daaelar** [input-leap/input-leap#1788] [Feature Requests]
In my use case, I have a bluetooth headset with touchpad controls on it. I would like an option in the server configuration UI to cause events from selected devices to only apply to the server and not be passed on to any selected clients. If this is a difficulty because of splitting event handling, then it would be better to at least just drop events from those devices when not controlling the server.

As an example: my bluetooth headset is connected to my linux server and playing music. I have a macOS client connected to this server. I would like the play/pause, back and next buttons on my headset to always be sent to the server even if I am currently actively using the macOS client. In current behavior, those input events are passed to the client, causing an (non-functional) Apple Music launch instead of pausing the current track in my linux mpris2 client.

Since many devices can be ephemeral connections such as the aforementioned headset or devices like security tokens, it would be good if there was a method to blacklist devices by HID type, known USB/PCI/Bluetooth bus addresses, or similar masking methods.

**Comment by shymega** [comment]
I think this is more related to the actual display server than Input Leap. We have a lot of work to do in refactoring event handling. I wouldn't expect anything for a while.
